client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -155,6 +156,15 @@
     */
    NotifyingFuture<Boolean> replaceWithVersionAsync(K key, V newValue, long version, int lifespanSeconds, int maxIdleSeconds);
 
+   /**
+    * Retrieve entries from the server
+    *
+    * @param filterConverterFactory Factory name for the KeyValueFilterConverter or null
+    * @param segments               The segments to iterate on or null if all segments should be iterated
+    * @param batchSize              The number of entries transferred from the server at a time
+    * @return Iterator for the entries
+    */
+   CloseableIterator<Entry<Object, Object>> retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize);
```

Comment:
I don't think the client itself should ever care about how to map keys to segments.  We should be able to do that internally ourselves.  The client however may want to do some sort of partitioning, in which case doing this by segment is a very logical thing.

It seems what we really need are these methods somewhere:
```
Set<Integer> getSegments()
Map<Address, Set<Integer>> getSegmentsByServer()
```

And then the RemoteCache would have the current method that @gustavonalle added.

Do we want another one like getAll but instead returns an iterator?  Something like:

```
CloseableIterator<Entry<K, Object>> retrieveEntries(String filterConverterFactory, Set<? extends K> keys, int batchSize);
```